db/compaction/compaction_iterator.cc

```diff
@@ -1081,15 +1081,18 @@ void CompactionIterator::GarbageCollectBlobIfNeeded() {
 
 void CompactionIterator::DecideOutputLevel() {
 #ifndef NDEBUG
-  // TODO: will be set by sequence number or key range, for now, it will only be
-  // set by unittest
+  // Could be overridden by unittest
   PerKeyPlacementContext context(level_, ikey_.user_key, value_,
                                  ikey_.sequence);
   TEST_SYNC_POINT_CALLBACK("CompactionIterator::PrepareOutput.context",
                            &context);
   output_to_penultimate_level_ = context.output_to_penultimate_level;
 #endif /* !NDEBUG */
 
+  if (ikey_.sequence > penultimate_level_cutoff_seqno_) {
+    output_to_penultimate_level_ = true;
+  }
```

Comment:
nit: combining this and line 1098 might provide slightly better performance and might be less confusing to some readers.

Reply:
udpated.